spark/src/main/scala/org/apache/spark/sql/delta/DeltaOperations.scala

```diff
@@ -491,17 +495,24 @@ object DeltaOperations {
   val OPTIMIZE_OPERATION_NAME = "OPTIMIZE"
   /** parameter key to indicate which columns to z-order by */
   val ZORDER_PARAMETER_KEY = "zOrderBy"
+  /** parameter key to indicate clustering columns */
+  val CLUSTERING_PARAMETER_KEY = "clusterBy"
 
   /** Recorded when optimizing the table. */
   case class Optimize(
       predicate: Seq[Expression],
       zOrderBy: Seq[String] = Seq.empty,
-      auto: Boolean = false
+      auto: Boolean = false,
+      clusterBy: Option[Seq[String]] = None
   ) extends OptimizeOrReorg(OPTIMIZE_OPERATION_NAME, predicate) {
     override val parameters: Map[String, Any] = super.parameters ++ Map(
       ZORDER_PARAMETER_KEY -> JsonUtils.toJson(zOrderBy),
+      CLUSTERING_PARAMETER_KEY -> JsonUtils.toJson(clusterBy.getOrElse(Seq.empty)),
       AUTO_COMPACTION_PARAMETER_KEY -> auto
-    )
+    ) ++ (if (clusterBy.nonEmpty) {
+      // When clustering columns are specified, we will clear the zOrderBy key.
+      Map(ZORDER_PARAMETER_KEY -> JsonUtils.toJson(Seq.empty))
```

Comment:
Given this is going to be OSSed, can we just fold this logic into the above like
```
override val parameters: Map[String, Any] = super.parameters ++ Map(
      ZORDER_PARAMETER_KEY -> if (clusterBy.nonEmpty) JsonUtils.toJson(Seq.empty) else JsonUtils.toJson(zOrderBy),
```